### PR TITLE
[TOOLS-4366] MariaDB feature

### PR DESCRIPTION
**Pull request by rathana-pvs**
### [[TOOLS-4366](http://jira.cubrid.org/browse/TOOLS-4366)] add maria db as a source db
Extend Cubrid Migration Tool to support MariaDB as a source database

### CODE REVIEW

File: com.cubrid.cubridmigration.core.testfragment/src/com/cubrid/cubridmigration/mariadb/MariaDB2CUBRIDMigParasTest.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2009 Search Solution Corporation. All rights reserved by Search Solution.
```
Comment (kisoo-han) [inline com.cubrid.cubridmigration.core.testfragment/src/com/cubrid/cubridmigration/mariadb/MariaDB2CUBRIDMigParasTest.java]: If we create new files, then we use copyright in license statements as:

` * Copyright (c) 2016 CUBRID Corporation.`

If we modified existing files, then we use copyright in license statements as:

```
 * Copyright (C) 2008 Search Solution Corporation.
 * Copyright (c) 2016 CUBRID Corporation.
```

File: com.cubrid.cubridmigration.core.testfragment/.classpath [hunk 2]
```diff
@@ -1,14 +1,19 @@
...
+	</classpathentry>
+	<classpathentry kind="con" path="org.eclipse.pde.core.requiredPlugins"/>
+	<classpathentry kind="lib" path="lib/cglib-nodep-2.1_3.jar"/>
+	<classpathentry kind="lib" path="lib/hamcrest-core-1.1.jar"/>
+	<classpathentry kind="lib" path="lib/hamcrest-library-1.1.jar"/>
+	<classpathentry kind="lib" path="lib/jmock-2.5.1.jar"/>
+	<classpathentry kind="lib" path="lib/jmock-junit4-2.5.1.jar"/>
+	<classpathentry kind="lib" path="lib/jmock-legacy-2.5.1.jar"/>
+	<classpathentry kind="lib" path="lib/objenesis-1.0.jar"/>
+	<classpathentry kind="lib" path="C:/Users/rathana/Downloads/slf4j-log4j12-1.6.4.jar"/>
```
Comment (kisoo-han) [inline com.cubrid.cubridmigration.core.testfragment/.classpath]: You referenced the file path C drive in your PC, codes of this style should be changed. And I'm not sure this file should be modified.

File: com.cubrid.cubridmigration.core.testfragment/src/com/cubrid/cubridmigration/mariadb/MariaDB2CUBRIDMigParasTest.java (hunk 1 above)
Comment (rathana-pvs) [reply com.cubrid.cubridmigration.core.testfragment/src/com/cubrid/cubridmigration/mariadb/MariaDB2CUBRIDMigParasTest.java]: Thank you for comment. I would like to confirm:
- create new file:
```
/*
 * Copyright (c) 2016 CUBRID Corporation. All rights reserved by Search Solution. 
```
 
 -  modify file.
```
/*
 * Copyright (C) 2008 Search Solution Corporation.
 * Copyright (C) 2009 Search Solution Corporation. All rights reserved by Search Solution. 
```
Am I right?

File: com.cubrid.cubridmigration.core.testfragment/.classpath (hunk 2 above)
Comment (rathana-pvs) [reply com.cubrid.cubridmigration.core.testfragment/.classpath]: Oh, Thank you for this suggestion. 
Yes, It should not referenced to my Pc.
I should copy it to lib folder in project and do  the build path again.

File: com.cubrid.cubridmigration.core.testfragment/src/com/cubrid/cubridmigration/mariadb/MariaDB2CUBRIDMigParasTest.java (hunk 1 above)
Comment (kisoo-han) [reply com.cubrid.cubridmigration.core.testfragment/src/com/cubrid/cubridmigration/mariadb/MariaDB2CUBRIDMigParasTest.java]: * when you create new file, copy existing license statement. And replace following line into new one

ASIS

`* Copyright (C) 2009 Search Solution Corporation. All rights reserved by Search Solution.`

TOBE

`* Copyright (c) 2016 CUBRID Corporation.`

-----

* when you **modify** existing file, Replace following line into new one

ASIS

`Copyright (C) 2009 Search Solution Corporation. All rights reserved by Search Solution.`

**TOBE**

```
* Copyright (C) 2008 Search Solution Corporation.
* Copyright (c) 2016 CUBRID Corporation.
```

File: com.cubrid.cubridmigration.core.testfragment/.classpath [hunk 3]
```diff
@@ -1,14 +1,19 @@
...
 	<classpathentry kind="src" path="src"/>
-	<classpathentry exported="true" kind="lib" path="lib/cglib-nodep-2.1_3.jar" sourcepath="com.cubrid.cubridmigration.core.testfragmentsrc.zip"/>
-	<classpathentry exported="true" kind="lib" path="lib/hamcrest-core-1.1.jar" sourcepath="com.cubrid.cubridmigration.core.testfragmentsrc.zip"/>
-	<classpathentry exported="true" kind="lib" path="lib/hamcrest-library-1.1.jar" sourcepath="com.cubrid.cubridmigration.core.testfragmentsrc.zip"/>
-	<classpathentry exported="true" kind="lib" path="lib/jmock-2.5.1.jar" sourcepath="com.cubrid.cubridmigration.core.testfragmentsrc.zip"/>
-	<classpathentry exported="true" kind="lib" path="lib/jmock-junit4-2.5.1.jar" sourcepath="com.cubrid.cubridmigration.core.testfragmentsrc.zip"/>
-	<classpathentry exported="true" kind="lib" path="lib/jmock-legacy-2.5.1.jar" sourcepath="com.cubrid.cubridmigration.core.testfragmentsrc.zip"/>
-	<classpathentry exported="true" kind="lib" path="lib/objenesis-1.0.jar" sourcepath="com.cubrid.cubridmigration.core.testfragmentsrc.zip"/>
+	<classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER/org.eclipse.jdt.internal.debug.ui.launcher.StandardVMType/JavaSE-17">
```
Comment (hwany7seo) [inline com.cubrid.cubridmigration.core.testfragment/.classpath]: The compiler for the release is jdk 1.8, And eclipse(3.2, indigo) for development are only supports 1.7 or lower.
Please keep the old version until we update eclipse and the build environment.

File: com.cubrid.cubridmigration.core.testfragment/.settings/org.eclipse.jdt.core.prefs [hunk 4]
```diff
@@ -1,8 +1,10 @@
-#Tue Dec 21 17:08:52 CST 2010
 eclipse.preferences.version=1
 org.eclipse.jdt.core.compiler.codegen.inlineJsrBytecode=enabled
-org.eclipse.jdt.core.compiler.codegen.targetPlatform=1.5
-org.eclipse.jdt.core.compiler.compliance=1.5
+org.eclipse.jdt.core.compiler.codegen.targetPlatform=17
```
Comment (hwany7seo) [inline com.cubrid.cubridmigration.core.testfragment/.settings/org.eclipse.jdt.core.prefs]: Please keep the jdk version in this file.

**Review by hwany7seo — APPROVED** [review]
Thank you for your efforts.

**Review by Kang-dot — APPROVED** [review]

**Review by kisoo-han — APPROVED** [review]
Thanks for your works and thanks for your first contribution to the CMT.

**Pull request merged**